service/src/models/archaeologist.ts

```diff
@@ -148,17 +148,23 @@ export class Archaeologist {
                 maxRewrapInterval,
                 diggingFee,
                 timestamp,
-              }: SarcophagusParameters = JSON.parse(new TextDecoder().decode(data));
+              }: SarcophagusNegotiationParams = JSON.parse(new TextDecoder().decode(data));
+
+              const maximumRewrapIntervalBN = BigNumber.from(maxRewrapInterval);
+
               let error: number | null = null;
-              if (maxRewrapInterval > inMemoryStore.profile!.maximumRewrapInterval.toNumber()) {
+              if (maximumRewrapIntervalBN.gt(inMemoryStore.profile!.maximumRewrapInterval)) {
```

Comment:
ahh I see it's a BigNumber on the profile too. I vote we make this a number and reserve BN for these 18 digit currency values we're dealing with